designs/2025-define-plugin/README.md

```diff
@@ -141,69 +132,354 @@ Object.assign(plugin.configs, {
       rules: Object.fromEntries(
         Object.entries(plugin.rules)
           .filter(([, rule]) => rule.meta.docs?.recommended)
-          .map(([ruleName]) => [ruleName, "error"]),
+          // (see #### Default Generated Config later for rule severities)
+          .map(([ruleName, rule]) => [ruleName, getRuleSeverity(rule)]),
       ),
     },
   ],
 });
 ```
 
 `definePlugin`'s design intentionally avoids referring to a `plugin` value.
-No `Object.assign()`, `Object.defineProperty()`, or `get()` should be required in the making of an ESLint plugin.
+No `Object.assign()`, `Object.defineProperty()`, or `get()` will be required in the making of an ESLint plugin with `definePlugin`.
 
 ### Configs
 
-`configs` is the only property `definePlugin` receives with a different shape.
-It is the same as before, but each config's `rules` must be provided either as:
+`definePlugin` will add two properties to all generated `configs` objects:
 
-- `null | undefined`: to preserve that value
-- An array containing any number of elements, each either:
-  - A rule object, in which case it will be turned into a namespaced rule based on the plugin's `meta.name` and the rule's key under the plugin's `rules`
-  - An array containing a rule object and a severity to specify that rule as
-  - An object with string keys and severity values, which will be directly merged into the resultant config `rules`
+- `name`: will default to the plugin's namespace + `"/"` + the config's key
+  - Plugin namespaces will be equal to the plugin's `meta.namespace ?? meta.name`
+- `plugin`: merged with an object defining the plugin under its name
 
-`definePlugin` will add two properties by default to config objects:
+`configs` is the only property `definePlugin` receives with a different shape than the output plugin object.
 
-- `name`: will default to the plugin's name + `"/"` + the config's key
-- `plugin`: merged with an object defining the plugin under its name
+- If `configs` is not provided, a default config is generated.
+- `configs` may be provided as an object whose properties describe the configs to create.
 
-The following definition redundantly defines its `recommended` config's `name` and `rules` the same as the earlier example plugin:
+The following two sections describe those two behaviors.
 
-```js
+#### Default Generated Config
+
+If `configs` is not provided, then `definePlugin` will create a `recommended` config with the following properties:
```

Comment:
What happens if no rules have `meta.docs.recommended`?

Reply:
Good point - I'd expect one wouldn't be created then?